src/contract.jl

```diff
@@ -73,7 +73,11 @@ function logscalar(
     denominator_terms
   end
 
-  return sum(log.(numerator_terms)) - sum(log.((denominator_terms)))
+  if any(t -> iszero(t), denominator_terms)
+    return -Inf
+  else
+    return sum(log.(numerator_terms)) - sum(log.((denominator_terms)))
+  end
```

Comment:
You removed `collect` but didn't simplify the code in the other ways that I suggested.